.github/workflows/publish-pypi.yml

```diff
@@ -0,0 +1,107 @@
+name: Publish Python 🐍 distribution 📦 to PyPI and TestPyPI
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  build:
+    name: Build distribution 📦
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: "3.10"
+    - name: Install pypa/build
+      run: python3 -m  pip install poetry wheel --user
+    - name: Build a binary wheel and a source tarball
+      run: python3 -m poetry build
+    - name: Store the distribution packages
+      uses: actions/upload-artifact@v3
+      with:
+        name: python-package-distributions
+        path: dist/
+
+  publish-to-pypi:
+    name: Publish Python 🐍 distribution 📦 to PyPI
+    if: startsWith(github.ref, 'refs/tags/')  # only publish to PyPI on tag pushes
```

Comment:
I don't think we push any other tags then the release tags but should someone push *any* tag it will push a release to pypi. Would it be safe to assume our tags always start with '20' and filter on 'refs/tags/20' to prevent a accidental push when someone tags eg dev-expiremental or whatever at some point....if too paranoid please ignore!